lua/autorun/server/sv_spawn_protection.lua

```diff
@@ -259,3 +265,26 @@ hook.Add( "KeyPress", "CFCspawnProtectionKeyPressCheck", spawnProtectionKeyPress
 
 -- Prevent entity damage while in spawn protection
 hook.Add( "EntityTakeDamage", "CFCpreventDamageDuringSpawnProtection", preventDamageDuringSpawnProtection, HOOK_HIGH )
+
+-- Disable weapon checking on a player once they die
+hook.Add( "PlayerDeath", "CFCspawnProtectionPauseCheckingWeapons", function( ply )
+    ply.spawnProtectionCannotCheckWeapons = true
+end )
+
+-- Resume weapon checking on a player with a slight delay after they spawn, after loadout addons give their weapons
+-- The expectation is for loadouts to prioritize holding the physgun at the end, making players not lose spawn protection
+-- Note that some loadout addons (like StyledStrike's) use a 0.1s timer on PlayerLoadout to give weapons
+hook.Add( "PlayerSpawn", "CFCspawnProtectionResumeCheckingWeapons", function( ply )
+    local playerIdentifer = playerStartCheckingWeaponsTimerIdentifier( ply )
+    timer.Create( playerIdentifer, 0.3, 1, function()
+        if not IsValid( ply ) then return end
+
+        ply.spawnProtectionCannotCheckWeapons = nil
+
+        local wep = ply:GetActiveWeapon()
+
+        if IsValid( wep ) then
+            spawnProtectionWeaponChangeCheck( ply, nil, wep )
```

Comment:
wont this just remove the spawnprotection after 0.3s? whats the difference with before here?

Reply:
cfc_pvp ensures players start out holding a physgun, so they won't lose it unless they switch off intentionally.

https://github.com/CFC-Servers/cfc_spawn_protection/assets/26103433/26ffd52a-6fc7-457d-8b01-271866f0c20e

Reply:
Actually no, turns out it's going to the physgun because I have it set as my preferred weapon in the loadout, which isn't a thing by default.
This might require a more complex solution or need some edits to StyledStrike's addon, like making it use the physgun as the preferred wep by default if none of the others are set as preferred.